### PR TITLE
embedding resolving callback

### PR DESCRIPTION
**Pull request by Amedeo91**
The purpose of this PR, is to provide some point of customization to the Framework consumer in order to put into place particular behavior as described per #1010.

This way, the framework will be more flexible and easier to put several uses cases (i.e. tree structure database) into place.

### CODE REVIEW

**Comment by Amedeo91** [comment]
Hi @nicolaiarocci, is there anything else I should do for this PR?

Regards,
Amedeo

File: docs/features.rst [hunk 1]
```diff
@@ -965,6 +965,71 @@ that are embedded by default:
...
+|Action |What    |When  |Event name / method signature                    |
++=======+========+======+=================================================+
+|Fetch  |Resource|Before|| ``on_embedding_resolving``                     |
+|       |        |      || ``def event(field_name, subdocument,           |
+|       |        |      ||             data_relation,subresources_query)``|
+|       +--------+------+-------------------------------------------------+
+|       |Item    |Before|| ``on_embedding_resolving_<resource_name>``     |
+|       |        |      || ``def event(subdocument, data_relation,        |
+|       |        |      ||             subresources_query)``              |
++-------+--------+------+-------------------------------------------------+
```
Comment (nicolaiarocci) [inline docs/features.rst]: The generic 'event hooks' table should also be updated with these new hooks

File: docs/features.rst [hunk 2]
```diff
@@ -965,6 +965,71 @@ that are embedded by default:
...
+
+.. code-block:: pycon
+
+   >>> def modify_query(field_name, subdocument, data_relation, subresources_query):
+   ...     subresources_query[field_name].update({"custom_id": subdocument["custom_id"]})
+
+   >>> app = Eve()
+   >>> app.on_embedding_resolving += modify_query
+
+You may use want to make the query result no results for a specific end-point:
```
Comment (nicolaiarocci) [inline docs/features.rst]: You probably mean "You may want to hook your callback to a specific resource endpoint instead. In the following example we are attaching a callback to the `products`  endpoint:"

File: eve/methods/common.py [hunk 3]
```diff
@@ -677,7 +677,7 @@ def resolve_embedded_fields(resource, req):
     return enabled_embedded_fields
 
 
-def embedded_document(references, data_relation, field_name):
+def embedded_document(last_field, subdocument, data_relation, field_name):
     """ Returns a document to be embedded by reference using data_relation
```
Comment (nicolaiarocci) [inline eve/methods/common.py]: Please update the docstring accordingly

File: docs/features.rst (hunk 2 above)
Comment (Amedeo91) [inline docs/features.rst]: Thanks!

**Comment by Amedeo91** [comment]
Hi @nicolaiarocci, I did the update.
I have a problem in squashing the commit because my commit history does not reflect the history of the main repository. Might you be able to do it directly once the PR is merged pls?

Regards,
Amedeo

File: eve/methods/common.py (hunk 3 above)
Comment (Amedeo91) [inline eve/methods/common.py]: update, thanks!

File: docs/features.rst (hunk 1 above)
Comment (Amedeo91) [inline docs/features.rst]: ok, done!

**Comment by nicolaiarocci** [comment]
Closing this for the time being.

**Comment by Amedeo91** [comment]
@nicolaiarocci Was it merged or not?

**Pull request closed without merging**